### PR TITLE
Generate async data sources by default.

### PR DESCRIPTION
**Pull request by CyrusNajmabadi**
fixes https://github.com/pulumi/tf2pulumi/issues/144

### CODE REVIEW

**Comment by CyrusNajmabadi** [comment]
> The async: true option should be driven off of the value of UsePromptDataSources.

Right you are.  PTAL and let me know if i did it correctly this time.  Note: i didn't drive off of UsePromptDataSources directly, but rather if the datasource resource was contained in `promptDataSources`.  That felt like the more correct way to do things, but i could be wrong here.

File: gen/nodejs/testdata/test_ordering/index_notprompt.ts [hunk 1]
```diff
@@ -0,0 +1,13 @@
...
+const config = new pulumi.Config();
+// Accept the AWS region as input.
+const awsRegion = config.get("awsRegion") || "us-west-2";
+
+// Create a provider for account data.
+const accountData = new aws.Provider("account_data", {
+    region: awsRegion,
+});
+// Get the caller's identity.
+const accountDataCallerIdentity = pulumi.output(aws.getCallerIdentity({provider: accountData,async: true}));
```
Comment (pgavlin) [inline gen/nodejs/testdata/test_ordering/index_notprompt.ts]: Can you file an issue for the lack of a space between `accountData` and `async: true`? I think this is a bug in nearby code.

Comment (CyrusNajmabadi) [reply gen/nodejs/testdata/test_ordering/index_notprompt.ts]: i'm happy to just fix it.  i noticed this oriinally but thought this was intentional :)

Comment (CyrusNajmabadi) [reply gen/nodejs/testdata/test_ordering/index_notprompt.ts]: (i will do in a separate PR)

Comment (CyrusNajmabadi) [reply gen/nodejs/testdata/test_ordering/index_notprompt.ts]: https://github.com/pulumi/tf2pulumi/issues/156

**Pull request merged**